package_check.sh

```diff
@@ -1,1518 +1,181 @@
-
-	# Default tests
-	pkg_linter=1
-	setup_sub_dir=1
-	setup_root=1
-	setup_nourl=0
-	upgrade=1
-	setup_private=1
-	setup_public=1
-	backup_restore=1
-	multi_instance=1
-	incorrect_path=0
-	port_already_use=0
-	change_url=0
-	all_test=$((all_test+9))
-
-
-	# Read each arguments and store them
-	while read line
-	do
-		# Read each argument and pick up the first value. Then replace : by =
-		add_arg="$(echo $line | cut -d ':' -f1,2 | sed s/:/=/)"
-		# Then add this argument and follow it by &
-		manifest_arguments="${manifest_arguments}${add_arg}&"
-	done < "$manifest_extract"
-
-	# Search a string in the partial check_process
-	find_string () {
-		echo $(grep "$1" "$manifest_extract")
-	}
-
-	# Try to find all specific arguments needed for the tests
-	keep_name_arg_only () {
-		# Find the line for the given argument
-		local argument=$(find_string "$1")
-		# If a line exist for this argument
-		if [ -n "$argument" ]; then
-			# Keep only the name of the argument
-			echo "$(echo "$argument" | cut -d ':' -f1)"
-		fi
-	}
-	domain_arg=$(keep_name_arg_only ":ynh.local")
-	path_arg=$(keep_name_arg_only "path:")
-	user_arg=$(keep_name_arg_only "user:\|admin:")
-	public_arg=$(keep_name_arg_only "is_public:")
-	# Find the values for public and private
-	if [ -n "$public_arg" ]
-	then
-		line=$(find_string "is_public:")
-		# Assume the first value is public and the second is private.
-		public_public_arg=$(echo "$line" | cut -d ":" -f2)
-		public_private_arg=$(echo "$line" | cut -d ":" -f3)
-	fi
-
-	count_test () {
-		# Decrease the number of test, if this test is not already removed.
-		if [ $1 -eq 1 ]; then
-			all_test=$((all_test-1))
-			return 1
-		fi
-	}
-
-	# Disable some tests if the manifest key doesn't be found
-	if [ -z "$domain_arg" ]
-	then
-		ECHO_FORMAT "The manifest key for domain didn't be find.\n" "lyellow"
-		setup_sub_dir=0
-		count_test "$setup_root" || setup_root=0
-		count_test "$multi_instance" || multi_instance=0
-		count_test "$incorrect_path" || incorrect_path=0
-		setup_nourl=1
-	fi
-	if [ -z "$path_arg" ]
-	then
-		ECHO_FORMAT "The manifest key for path didn't be find.\n" "lyellow"
-		count_test "$setup_root" || setup_root=0
-		count_test "$multi_instance" || multi_instance=0
-		count_test "$incorrect_path" || incorrect_path=0
-	fi
-	if [ -z "$public_arg" ]
-	then
-		ECHO_FORMAT "The manifest key for public didn't be find.\n" "lyellow"
-		setup_private=0
-		setup_public=0
-		all_test=$((all_test-2))
-	fi
-	# Remove the multi-instance test if this parameter is set at false in the manifest.
-	if grep multi_instance "$package_path/manifest.json" | grep -q false
-	then
-		count_test "$multi_instance" || multi_instance=0
-	fi
-
-	# Launch all tests successively
-	TESTING_PROCESS
-	# Print the final results of the tests
-	TEST_RESULTS
+    package_check_pid="$(cat "./pcheck.lock" 2> /dev/null | cut -d: -f3)"
+    if [ -n "$package_check_pid" ]; then
+        kill --signal 15 $package_check_pid
+    fi
+    clean_exit 0
```

Comment:
We don't really want to exit the script here, just clean things up
```suggestion
    LXC_RESET

    [ -n "$TEST_CONTEXT" ] && rm -rf "$TEST_CONTEXT"
    rm -f "$lock_file"
```

Reply:
weeeell in fact this option is only meant to interface with CI_package_check which wants to clean things up if package_check times out ...

Reply:
ho we're not supposed to use it with a package to test as agrument?

Reply:
eeeeh not really, at least that wasn't the intention xD

Reply:
Ok! :-)